File: docs/guides/python/saveImageClassifications.py
from algoliasearch.search.client import SearchClientSync

from algoliasearch.search.models.browse_response import BrowseResponse


def _get_image_labels(image_url, object_id, score_limit):
    # Implement your image classification logic here
    return {"objectID": "", "imageURL": "", "objects": []}


_client = SearchClientSync("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")


records = []


def _aggregator(res: BrowseResponse):
    for hit in res.hits:
        props = hit.to_dict()
        image_url = props["imageURL"]
        records.append(_get_image_labels(image_url, hit.object_id, 0.5))


_client.browse_objects("<YOUR_INDEX_NAME>", aggregator=_aggregator)

_client.partial_update_objects(
    index_name="<YOUR_INDEX_NAME>",
    objects=records,
    create_if_not_exists=True,
)

File: docs/guides/go/src/saveImageClassifications.go
package main

import (
	"github.com/algolia/algoliasearch-client-go/v4/algolia/search"
)

func saveImageClassifications() {
	type Image struct {
		ImageURL string           `json:"imageURL"`
		ObjectID string           `json:"objectID"`
		Objects  []map[string]any `json:"objects"`
	}

	getImageLabels := func(imageURL, objectID string, _scoreLimit float64) Image {
		// Implement your image classification logic here
		return Image{
			ImageURL: imageURL,
			ObjectID: objectID,
			Objects:  []map[string]any{},
		}
	}

	client, err := search.NewClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		panic(err)
	}

	images := []Image{}

	err = client.BrowseObjects("<YOUR_INDEX_NAME>", search.BrowseParamsObject{}, search.WithAggregator(func(res any, err error) {
		if err != nil {
			panic(err)
		}

		browseRes, ok := res.(search.BrowseResponse)
		if !ok {
			panic("Invalid response")
		}

		for _, hit := range browseRes.Hits {
			props := hit.AdditionalProperties
			imageURL, _ := props["imageURL"].(string)
			images = append(images, getImageLabels(imageURL, hit.GetObjectID(), 0.5))
		}
	}))
	if err != nil {
		panic(err)
	}

	records := make([]map[string]any, len(images))
	for i, img := range images {
		records[i] = map[string]any{
			"imageURL": img.ImageURL,
			"objectID": img.ObjectID,
			"objects":  img.Objects,
		}
	}

	_, err = client.PartialUpdateObjects(
		"<YOUR_INDEX_NAME>", records, search.WithCreateIfNotExists(true))
	if err != nil {
		panic(err)
	}
}
